### ISSUE TITLE
Sass @extend definition referencing parent selectors (&)

### ISSUE THREAD

**Issue by huyby** [sass/sass#848]
I'm trying to accomplish the following CSS output with Sass:

```
.selector1 + .selector1, .selector2 + .selector2 {
    margin-top: -80px;
}
```

The @extend directive should do the job:

```
%sibling-selector {
    & + & {
        margin-top: -80px;  
    }
}
.selector1 {
    @extend %sibling-selector;
}
.selector2 {
    @extend %sibling-selector;
}
```

But it seems @extend has problems with the two parent references (&) in the @extend definition (%sibling-selector).
This is the output:

```
.selector1 + .selector1, .selector2 + .selector1, 
.selector1 + .selector2, .selector2 + .selector2 {
    margin-top: -80px;
}
```

The @extend directive is making sibling selectors for each combination of selectors that use the @extend definition.

While I would expect the @extend directive stays in the "scope" of the current selector, and both ampersands are replaced by the current selector and not referring to previous ones.

**Comment by chriseppstein** [comment]
Let's consider what this really means:

``` scss
%sibling-selector {
    & + & {
        margin-top: -80px;  
    }
}
```

It says that any two elements that are a `%sibling-selector` and are next to each other should get a negative margin.

Now, you've said that `.selector1` is a `%sibling-selector` and that `.selector2` is a `%sibling-selector`. So sass has to permute the possibilities in order to satisfy the legal documents that you may create given this declaration.

Also, please keep in min that this is not going to do what you want it to if you have any nested selectors that are extending %sibling-selector. In fact, it becomes quite hard to build this in a generic way. However it will be possible in sass 3.3 thanks to #286.

**Comment by filip-jk** [comment]
Hi @chriseppstein 

Apologize for re-opening this but my issue is strictly related to this one. In your rseponse, you mention that _However it will be possible in sass 3.3 thanks to #286_ I did check that issue but I couldn't find a solution to this specific case. Would you mind telling how the original issue posted here by @huyby could be resolved/achieved in sass 3.3? Thank you so much!
